posthog/models/experiment.py

```diff
@@ -62,3 +65,27 @@ class ExperimentHoldout(models.Model):
     created_by = models.ForeignKey("User", on_delete=models.SET_NULL, null=True)
     created_at = models.DateTimeField(default=timezone.now)
     updated_at = models.DateTimeField(auto_now=True)
+
+
+class ExperimentSavedMetric(models.Model):
+    name = models.CharField(max_length=400)
+    description = models.CharField(max_length=400, null=True, blank=True)
+    team = models.ForeignKey("Team", on_delete=models.CASCADE)
+
+    query = models.JSONField()
+
+    created_by = models.ForeignKey("User", on_delete=models.SET_NULL, null=True)
+    created_at = models.DateTimeField(default=timezone.now)
+    updated_at = models.DateTimeField(auto_now=True)
+
+
+class ExperimentToSavedMetric(models.Model):
```

Comment:
Why do we need this model?

I guess it has to do with the fact that we need to store different metadata for a metric for a given experiment.

Just wondering if it could be easier to handle it like this:
```
{
    "display": "primary",
    "type": "query",
    "query": <ExperimentTrendsQuery>
},
{
    "display": "secondary",
    "type": "saved_metric",
    "saved_metric_id": <id>
}
```

Reply:
if there's any metadata involved we need a custom many to many mapping table. What you're suggesting is basically us handling all of deletions / consistency with metrics because this wouldn't be a db level constraint, but a virtual constraint that we need to manage ourselves in the metrics json, which can get a bit painful 😅  (ex: When a saved metric is deleted we'd need to go through all experiments, check the metrics, and remove any records with the id)

Reply:
okay, well this is just my lack of experience with Django showing 😅 

makes perfect sense 👍